Assets/Countly/Plugins/CountlySDK/Services/UserProfile.cs

```diff
@@ -0,0 +1,682 @@
+    {
+        try {
+            if (!(value is double || value is int || value is string)) {
+                Console.WriteLine("[UserProfile][ModifyCustomData] Provided an unsupported type for 'value'");
+                return;
+            }
+
+            object truncatedValue;
+            string truncatedKey = TrimKey(key);
+
+            if (value is string stringValue) {
+                truncatedValue = TrimValue(truncatedKey, stringValue);
+            } else {
+                truncatedValue = value;
+            }
+
+            CustomMods ??= new Dictionary<string, JObject>();
+
+            JObject ob;
+            if (mod != "$pull" && mod != "$push" && mod != "$addToSet") {
+                ob = new JObject { [mod] = JToken.FromObject(truncatedValue) };
+            } else {
+                if (CustomMods.ContainsKey(truncatedKey)) {
+                    ob = CustomMods[truncatedKey];
+                } else {
+                    ob = new JObject();
+                }
+
+                if (ob[mod] is JArray existingArray) {
+                    existingArray.Add(JToken.FromObject(truncatedValue));
+                } else {
+                    ob[mod] = new JArray(truncatedValue);
+                }
+            }
+            CustomMods[truncatedKey] = ob;
+        } catch (Exception e) {
+            Log.Error($"[UserProfile][ModifyCustomData] An exception occured during modifying the custom data. Exception: {e}");
+        }
+    }
+
+    private void ClearInternal()
+    {
+        Log.Debug("[UserProfile][ClearInternal] Start");
+
+        Name = null;
+        Username = null;
+        Email = null;
+        Organization = null;
+        Phone = null;
+        PictureUrl = null;
+        Gender = null;
+        BirthYear = 0;
+        CustomDataProperties = new Dictionary<string, object>();
+        CustomMods = new Dictionary<string, JObject>();
+    }
+
+    private string ConvertToJSON()
+    {
+        JObject json = new JObject();
+
+        try {
+            if (Name != null) {
+                if (Name == "") {
+                    json.Add(NAME_KEY, null);
+                } else {
+                    json.Add(NAME_KEY, Name);
+                }
+            }
+
+            if (Username != null) {
+                if (Username == "") {
+                    json.Add(USERNAME_KEY, null);
+                } else {
+                    json.Add(USERNAME_KEY, Username);
+                }
+            }
+
+            if (Email != null) {
+                if (Email == "") {
+                    json.Add(EMAIL_KEY, null);
+                } else {
+                    json.Add(EMAIL_KEY, Email);
+                }
+            }
+
+            if (Organization != null) {
+                if (Organization == "") {
+                    json.Add(ORG_KEY, null);
+                } else {
+                    json.Add(ORG_KEY, Organization);
+                }
+            }
+

```

Comment:
this string checks are looking same. MB we can extract them to a function
like
it takes value, jobj and key. And adds it to the jobj if conditions met